### PR TITLE
Added support for excluded_symbol_names  to FileNameMustMatchTypeNameAnalyzer (MA0048)

### LINKED ISSUE #602: Can MA0048 support excluded_symbol_names with wildcard?
**Is your feature request related to a problem? Please describe.**

I'd like to permit only certain classes to be collocated within the same file:

```c#
public class FooHandler {}

public class FooCommand {}
```

or

```c#
public class Bar {}

public static class BarExtensions {}
```

**Describe the solution you'd like**

I tried to adjust the rule using editorconfig;

```
dotnet_diagnostic.MA0048.excluded_symbol_names = *Extensions|*Command|*Query
```

But the build still produces a warning;

```
Warning
MA0048
File name must match type name (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0048.md)
```

**Describe alternatives you've considered**

None, please suggest if there is any.

**Additional context**

```xml
<PackageReference Include="Meziantou.Analyzer" Version="2.0.87">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```

### PR DESCRIPTION
**Pull request by abatishchev**
Resolves #602.

Checklist:

- [x] Update FileNameMustMatchTypeNameAnalyzer .cs
- [x] Add tests
- [x] Update docs

### CODE REVIEW

**Comment by abatishchev** [comment]
@meziantou please take a look at the code change and the tests added

File: src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs [hunk 1]
```diff
@@ -44,14 +45,35 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context)
...
+            // dotnet_diagnostic.MA0048.excluded_symbol_names
+            var excludedSymbolNames = context.Options.GetConfigurationValue(location.SourceTree, "dotnet_diagnostic." + s_rule.Id + ".excluded_symbol_names", defaultValue: string.Empty);
+            if (!string.IsNullOrEmpty(excludedSymbolNames))
+            {
+                var matched = false;
+
+                var excludedSymbolNamesSplit = excludedSymbolNames.Split('|', StringSplitOptions.RemoveEmptyEntries);
+                foreach (var excludedSymbolName in excludedSymbolNamesSplit)
+                {
+                    if (IsWildcardMatch(symbolName, excludedSymbolName))
```
Comment (meziantou) [inline src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs]: The logic from similar rules is a little bit more complicated. For instance, https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508?WT.mc_id=DT-MVP-5003978#exclude-specific-symbols

> - Symbol name only (includes all symbols with the name, regardless of the containing type or namespace).
> - Fully qualified names in the symbol's [documentation ID format](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/#id-strings). Each symbol name requires a symbol-kind prefix, such as M: for methods, T: for types, and N: for namespaces.

It seems you only handle the first case. You could also support `T:MyNamespace.Dummy*`. Note that you can generate the documentation id using `DocumentationCommentId.CreateDeclarationId`

**Comment by abatishchev** [comment]
Sorry, I was out of town, will come back to the change this week.

File: src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs (hunk 1 above)
Comment (abatishchev) [reply src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs]: @meziantou done, please review.

Comment (meziantou) [reply src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs]: Good! Can you also update the documentation before I can merge? (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0048.md)

Comment (abatishchev) [reply src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs]: Updated the doc

**Comment by abatishchev** [comment]
Awesome, thanks! I enjoyed working in your codebase and learned something new along the way.

Let me know if I can try out a pre-release version, or if maybe you could publish the latest release?

**Comment by meziantou** [comment]
Every commit on the main branch are published to nuget.org. The new package should be available in a few minutes :)

**Pull request merged**